playground/backend/internal/code_processing/code_processing.go

```diff
@@ -235,6 +239,52 @@ func cancelCheck(ctx context.Context, pipelineId uuid.UUID, cancelChannel chan b
 	}
 }
 
+// readLogFile reads logs from the log file and keeps it to the cache.
+func readLogFile(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycle, pipelineId uuid.UUID, stopReadLogsChannel, finishReadLogChannel chan bool) {
```

Comment:
Also look like a lot of arguments... From the description of the method it looks that we need only:
1. log file
2. cache object
3. additional structure like context

Reply:
Changed.